x-pack/plugins/task_manager/server/task_pool.ts

```diff
@@ -158,3 +169,11 @@ function partitionListByCount<T>(list: T[], count: number): [T[], T[]] {
   const listInCount = list.splice(0, count);
   return [listInCount, list];
 }
+
+function isTaskSavedObjectNotFoundError(err: Error | Boom, taskId: string) {
+  return (
+    Boom.isBoom(err) &&
```

Comment:
IIRC, we're planning on de-Boom-ifying alerting plugins, so hopefully when that happens we can figure out a better story here.  Worth a comment, I think.

One theoretic problem with this today is that pre-registered actions have user-specified ids, so it's possible someone could game the system with a carefully crafted action id.  In practice, I believe the id's here are prefixed with `action:` or such anyway, so I think it's safe, and the worse case is they'd see a message we're trying to hide?

Reply:
Yeah, this is definitely not going to work once Boom errors are removed.
We're kind of counting on the fact that by using SavedObject's helper functions in the tests - it'll force a conversation with Platform when they remove that code, which will trigger us to find another solution.

I can also add a comment, but I counted on us using their code to be a more reliable trigger than a comment that will only become relevant when we're trying to figure out what has already broken. :)

Reply:
I've refactored it a bit - we now use a unit test to ensure that the underlying error created by the Saved Objects helper actually parses as a string format that contains the id correctly.
This way, if SO remove Boom errors, this will still work.
But if they change the format - it will break on CI and force the conversation. :)